### PR TITLE
Feature/apps 2455 role filter

### PR DESCRIPTION
**Pull request by solomontang**
# Feature
- Adds a text input to filter roles
image

### CODE REVIEW

**Review by anzioka — DISMISSED** [review]

**Comment by anzioka** [comment]
debounce 🚀

File: src/containers/org/members/RolesSection.js [hunk 1]
```diff
@@ -60,26 +68,24 @@ const RolesSection = ({
...
-              label={role.title}
-              mode="button"
-              onChange={handleOnChangeRoleCheckBox} />
-        ))
-      }
-      {
-        isOwner && isVisibleAddRoleToOrgModal && (
+        isOwner && (
           <AddRoleToOrgModal
+              isVisible={isVisibleAddRoleToOrgModal}
```
Comment (UnsungHero97) [inline src/containers/org/members/RolesSection.js:88]: Why are you making this change?

File: src/containers/org/members/FilteredRoles.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const FilteredRoles = ({
+  filterTerm,
+  onChange,
+  roles,
+  selectedRoleId,
+} :Props) => {
+
+  let filteredRoles = roles;
+  if (filterTerm) {
+    filteredRoles = filteredRoles.filter((role) => role.title.toLowerCase().includes(filterTerm));
```
Comment (UnsungHero97) [inline src/containers/org/members/FilteredRoles.js:23]: This bug got me in the orgs filter... you'll want to `.toLowerCase()` the filter term as well.

File: src/containers/org/members/RolesSection.js [hunk 3]
```diff
@@ -2,12 +2,15 @@
...
 
-import React, { useState } from 'react';
+import React, { useCallback, useState } from 'react';
 
 import styled from 'styled-components';
-import { Button, Checkbox } from 'lattice-ui-kit';
+import { Button, Checkbox, SearchInput } from 'lattice-ui-kit';
 import { ValidationUtils } from 'lattice-utils';
-import type { Organization, Role, UUID } from 'lattice';
+import { debounce } from 'lodash';
```
Comment (UnsungHero97) [inline src/containers/org/members/RolesSection.js:10]: There was something about lodash and webpack tree shaking... please import as follows:

```
import _debounce from 'lodash/debounce';
```

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

File: src/containers/org/members/RolesSection.js (hunk 1 above)
Comment (solomontang) [reply src/containers/org/members/RolesSection.js:88]: To fix the transition. The modal should always be mounted.

**Review by UnsungHero97 — APPROVED** [review]

**Review by anzioka — APPROVED** [review]

**Pull request merged**